### PR TITLE
CACTUS-428 :: IconButton Focus

### PR DESCRIPTION
**Pull request by Dhalton**
Ticket: https://repayonline.atlassian.net/browse/CACTUS-428

It kind of sucks that I ended up having to make changes to `DateInput` in order to get the integration tests to pass, but as far as I can tell the changes aren't causing problems. I'm not completely sure why adding an `:after` pseudo element caused tests to fail in FireFox in the first place, but it did, and that's how I was able to get around it. 🤷‍♂️ 

### Testing
1. Bring up the `IconButton` story and focus on the button to make sure the outline appears.
2. Change the `IconButton` size and the shape in the theme and make sure it looks good with different shapes & sizes.
3. Repeat steps 1 & 2 in IE11 to double check that it works there too, because I did have to do a workaround for IE.
4. Go to the `DateInput` story and make sure that works too -- specifically when you click the calendar icon to open the popup.

### CODE REVIEW

File: modules/cactus-web/src/DateInput/DateInput.tsx [hunk 1]
```diff
@@ -983,7 +983,7 @@ class DateInputBase extends Component<DateInputProps, DateInputState> {
   }
 
   private handleClick = (): void => {
-    if (this._didClickButton) this._open()
+    //if (this._didClickButton) this._open()
```
Comment (wilysword) [inline modules/cactus-web/src/DateInput/DateInput.tsx:986]: Should you just remove this?

**Review by wilysword — APPROVED** [review]
I like it. Worked everywhere I tested except Safari (because the stupid browser doesn't allow you to focus buttons).

Completely unrelated, but the behavior of "Second Accordion" in https://github.com/repaygithub/cactus/pull/440/files?short_path=4080036#diff-40800369d687a4a9cd456611609b8385d09c9f18aa571e0cf366b6a35aa6b9bc is kind of iffy, we might want to make a ticket for that.

**Review by mikesoltow — COMMENTED** [review]
Looks pretty good to me.  Just a few questions.

**Comment by mikesoltow** [comment]
> Completely unrelated, but the behavior of "Second Accordion" in https://github.com/repaygithub/cactus/pull/440/files?short_path=4080036#diff-40800369d687a4a9cd456611609b8385d09c9f18aa571e0cf366b6a35aa6b9bc is kind of iffy, we might want to make a ticket for that.

Agreed, although it looks like it's an issue of the manual formatting we put into the story rather than something baked into the Accordion component itself, unless I'm missing something.

File: modules/cactus-web/src/IconButton/IconButton.tsx [hunk 2]
```diff
@@ -110,18 +123,41 @@ export const IconButton = styled(IconButtonBase)<IconButtonProps>`
...
+      ${(p) => {
+        if (isIE) {
+          const IEOffset = focusOutlineSpacing[p.iconSize || 'medium'] / 2
+          return `
+            top: -${IEOffset}px;
+            left: -${IEOffset}px;
+          `
+        }
+      }}
+      border: 1px solid;
```
[the remark this answers is not in the document]
Comment (Dhalton) [reply modules/cactus-web/src/IconButton/IconButton.tsx:156]: Yeah, I checked with Hector on this and he said we should match the regular button behavior so I'll update this.

File: modules/cactus-web/src/IconButton/IconButton.tsx [hunk 3]
```diff
@@ -110,18 +123,41 @@ export const IconButton = styled(IconButtonBase)<IconButtonProps>`
   box-sizing: border-box;
   align-items: center;
   justify-content: center;
-  padding: 1px;
+  padding: 0px;
```
[the remark this answers is not in the document]
Comment (Dhalton) [reply modules/cactus-web/src/IconButton/IconButton.tsx:126]: It kept throwing off the alignment, mainly in IE. I can try to make it work without removing that, but I don't know why it's there in the first place.

File: modules/cactus-web/src/IconButton/IconButton.tsx [hunk 4]
```diff
@@ -81,6 +81,19 @@ const disabled: FlattenInterpolation<ThemeProps<CactusTheme>> = css`
   cursor: not-allowed;
 `
 
+const shapeMap = {
+  square: 'border-radius: 1px;',
+  intermediate: 'border-radius: 8px;',
+  round: 'border-radius: 50%;',
+}
```
[the remark this answers is not in the document]
Comment (Dhalton) [reply modules/cactus-web/src/IconButton/IconButton.tsx:88]: Yeah, I had to set it to 50% to make sure it'd be a circle no matter what the icon size is. I can let him know that it'll need to be updated

[the remark this answers is not in the document]
Comment (mikesoltow) [reply modules/cactus-web/src/IconButton/IconButton.tsx:89]: :+1:

**Review by mikesoltow — APPROVED** [review]
Changes look good...thanks!

**Pull request merged**